test/auth/oidc_test.go

```diff
@@ -52,3 +54,21 @@ func TestBrokerSupportsOIDC(t *testing.T) {
 	env.TestSet(ctx, t, oidc.AddressableOIDCConformance(broker.GVR(), "Broker", name, env.Namespace()))
 	env.Test(ctx, t, oidc.BrokerSendEventWithOIDCToken())
 }
+
+func TestSequenceSupportsOIDC(t *testing.T) {
+	t.Parallel()
+
+	ctx, env := global.Environment(
+		knative.WithKnativeNamespace(system.Namespace()),
+		knative.WithLoggingConfig,
+		knative.WithTracingConfig,
+		k8s.WithEventListener,
+		environment.Managed(t),
+		environment.WithPollTimings(4*time.Second, 12*time.Minute),
+	)
+
+	name := feature.MakeRandomK8sName("sequence")
+	env.Prerequisite(ctx, t, sequencefeatures.GoesReady(name))
+
+	env.Test(ctx, t, oidc.AddressableHasAudiencePopulated(sequence.GVR(), sequence.GVK().Kind, name, env.Namespace()))
```

Comment:
```
--- FAIL: TestSequenceSupportsOIDC/Sequence_populates_its_.status.address.audience_correctly (0.03s)
        --- FAIL: TestSequenceSupportsOIDC/Sequence_populates_its_.status.address.audience_correctly/Assert (0.00s)
            --- FAIL: TestSequenceSupportsOIDC/Sequence_populates_its_.status.address.audience_correctly/Assert/Sequence_have_audience_set (0.00s)
                addressable.go:67: audience of address (&{<nil> http://sequence-orcfqqgm-kn-sequence-0-kn-channel.test-vexahjlo.svc.c6668628035.local/ <nil> 0xc00068b910}) does not match expected audience flows.knative.dev/sequence/test-vexahjlo/sequence-orcfqqgm
```

Sorry @md-saif-husain I missed something: Seems we can't reuse the `AddressableHasAudiencePopulated` here directly, as we use the underlying channels audience as the audience of the sequence.
In the initial broker e2e test, we checked the audience directly, so you can check on this too: https://github.com/knative/eventing/pull/7237/commits/7af5d67de90ece0b9551c353cb97d967248b951b, where in your case the expectedAudience, comes from the channel (`channel_impl.GVR()`).

Reply:
> ```
> --- FAIL: TestSequenceSupportsOIDC/Sequence_populates_its_.status.address.audience_correctly (0.03s)
>         --- FAIL: TestSequenceSupportsOIDC/Sequence_populates_its_.status.address.audience_correctly/Assert (0.00s)
>             --- FAIL: TestSequenceSupportsOIDC/Sequence_populates_its_.status.address.audience_correctly/Assert/Sequence_have_audience_set (0.00s)
>                 addressable.go:67: audience of address (&{<nil> http://sequence-orcfqqgm-kn-sequence-0-kn-channel.test-vexahjlo.svc.c6668628035.local/ <nil> 0xc00068b910}) does not match expected audience flows.knative.dev/sequence/test-vexahjlo/sequence-orcfqqgm
> ```
> 
> 
>     
>   
> 
> Sorry @md-saif-husain I missed something: Seems we can't reuse the `AddressableHasAudiencePopulated` here directly, as we use the underlying channels audience as the audience of the sequence.
> In the initial broker e2e test, we checked the audience directly, so you can check on this too: [7af5d67](https://github.com/knative/eventing/commit/7af5d67de90ece0b9551c353cb97d967248b951b), where in your case the expectedAudience, comes from the channel (`channel_impl.GVR()`).

Please help me with how to setup channel with audience to test this e2e. Some pointer or reference would be appreciated.

I am trying something like this but audience of channel is not set . I may be missing something .
```

	ctx, env := global.Environment(
		knative.WithKnativeNamespace(system.Namespace()),
		knative.WithLoggingConfig,
		knative.WithTracingConfig,
		k8s.WithEventListener,
		environment.Managed(t),
		environment.WithPollTimings(4*time.Second, 12*time.Minute),
	)
	sequenceName := feature.MakeRandomK8sName("sequence")
	template := channel_template.ImmemoryChannelTemplate()

	env.Prerequisite(ctx, t, sequencefeatures.GoesReady(sequenceName, sequence.WithChannelTemplate(template)))

	env.Test(ctx, t, oidc.SequenceGetsAudiencePopulated(sequenceName, env.Namespace()))
```

This is e2e test for sequence audience 
```

	  expectedAudience := auth.GetAudience(channel_impl.GVR().GroupVersion().WithKind("InMemoryChannel"), v1.ObjectMeta{
		  Name:      name,
		  Namespace: namespace,
	  })
  
	  f.Alpha("Sequence").Must("have audience set", sequence.ValidateAddress(name, addressable.AssertAddressWithAudience(expectedAudience)))
  
	  return f
```

Reply:
Hey @md-saif-husain,
make sure that the authentication-oidc flag is enabled when running the test (hint: it was renamed from `authentication.oidc` to `authentication-oidc` in #7422).

Then I would create a new e2e test feature, which sets up a sequence (you can check on https://github.com/knative/eventing/blob/d978f3c993fdf99b4c5491f667a1b96ea6ca75d6/test/rekt/features/sequence/feature.go#L39 for example) and then check that the audience fits (as you have done with the `expectedAudience` bits).

Then your e2e test feature gets called in an acutall TestXYZ function (which has the `global.Environment` setup etc.). This is how we usually do it (1. create a feature and 2. call this feature in an actual e2e test).

I am currently traveling and on a conference. If the above doesn't help you, I can check if I can you provide a more concrete example by mid/end of this week.

Reply:
Hey @md-saif-husain,
In https://github.com/knative/eventing/commit/17572031cad40b45c267d543fd56ba6ac8904a69, I added the test on top of your PR. If you want, I can check if I can push it directly into your branch

Reply:
@md-saif-husain: Seems I can't push to your branch. Can you cherry-pick or copy it by yourself?